In [1]:
import pandas as pd
import nltk
traintext_data = pd.read_csv('traintextual.csv')
trainintuitive_data = pd.read_csv('trainintuitive.csv')

In [2]:
print('Shape of Train data:')
print('Textual data',traintext_data.shape)
print('Inuitive data',trainintuitive_data.shape)


Shape of Train data:
Textual data (9655, 4)
Inuitive data (8621, 4)


In [3]:
traintext_data.head(10)

source    name  id judgment
0  textual  Asthma   1        U
1  textual  Asthma   2        Y
2  textual  Asthma   4        U
3  textual  Asthma   6        U
4  textual  Asthma  13        U
5  textual  Asthma  15        U
6  textual  Asthma  16        U
7  textual  Asthma  19        U
8  textual  Asthma  20        U
9  textual  Asthma  21        U

In [4]:
trainintuitive_data.head(10)

source    name  id judgment
0  intuitive  Asthma   1        N
1  intuitive  Asthma   2        Y
2  intuitive  Asthma   4        N
3  intuitive  Asthma   6        N
4  intuitive  Asthma  15        N
5  intuitive  Asthma  16        N
6  intuitive  Asthma  19        N
7  intuitive  Asthma  20        N
8  intuitive  Asthma  21        N
9  intuitive  Asthma  22        N

In [5]:
len(traintext_data)

9655

In [6]:
len(trainintuitive_data)

8621

In [7]:
traintext_data['name'].value_counts()

OSA                     609
Hypertriglyceridemia    609
PVD                     609
Gallstones              607
Gout                    607
Venous Insufficiency    606
Asthma                  606
Obesity                 606
Depression              605
OA                      603
GERD                    603
Diabetes                601
Hypertension            601
Hypercholesterolemia    598
CAD                     594
CHF                     591
Name: name, dtype: int64

In [8]:
trainintuitive_data['name'].value_counts()

Gout                    598
OSA                     597
Gallstones              593
Hypertriglyceridemia    587
Depression              582
Diabetes                572
Asthma                  572
OA                      566
PVD                     557
Obesity                 554
CAD                     552
Hypertension            531
Venous Insufficiency    526
Hypercholesterolemia    503
GERD                    488
CHF                     243
Name: name, dtype: int64

In [9]:
traintext_data.nunique()

source        1
name         16
id          611
judgment      4
dtype: int64

The above stat states that ...we have
- 16 names ie. morbidity conditions
- 611 id i.e. 611 unique patient records
- 4 judgement - {Y,N,U,Q}

In [10]:
trainintuitive_data.nunique()

source        1
name         16
id          609
judgment      3
dtype: int64

The above stat states that ...we have
- 16 names ie. morbidity conditions
- 609 id i.e. 609 unique patient records
- 3 judgement - {Y,N,Q}

In [11]:
print("Features in the datasets :\n",'-' * 40)
print(list(traintext_data.columns))
print('\n')
print("Number of observations in the datasets :\n",'-' * 40)
print(len(traintext_data))


Features in the datasets :
 ----------------------------------------
['source', 'name', 'id', 'judgment']


Number of observations in the datasets :
 ----------------------------------------
9655


In [12]:
print("Features in the datasets :\n",'-' * 40)
print(list(trainintuitive_data.columns))
print('\n')
print("Number of observations in the datasets :\n",'-' * 40)
print(len(trainintuitive_data))

Features in the datasets :
 ----------------------------------------
['source', 'name', 'id', 'judgment']


Number of observations in the datasets :
 ----------------------------------------
8621


In [13]:
print("Datatypes of features in the datasets :\n",'-' * 40)
print(traintext_data.dtypes)


Datatypes of features in the datasets :
 ----------------------------------------
source      object
name        object
id           int64
judgment    object
dtype: object


In [14]:
print("Datatypes of features in the datasets :\n",'-' * 40)
print(trainintuitive_data.dtypes)


Datatypes of features in the datasets :
 ----------------------------------------
source      object
name        object
id           int64
judgment    object
dtype: object


# DATA PREPROCESSING

In [15]:
df_text1 = traintext_data[(traintext_data['judgment'] == 'Y') | (traintext_data['judgment']=='N')]

In [16]:
type(traintext_data)

pandas.core.frame.DataFrame

In [17]:
df_text1

source                  name    id judgment
1     textual                Asthma     2        Y
11    textual                Asthma    23        Y
15    textual                Asthma    28        Y
21    textual                Asthma    47        Y
33    textual                Asthma    69        Y
...       ...                   ...   ...      ...
9375  textual  Venous Insufficiency   879        Y
9463  textual  Venous Insufficiency   989        Y
9511  textual  Venous Insufficiency  1055        Y
9578  textual  Venous Insufficiency  1149        Y
9631  textual  Venous Insufficiency  1216        Y

[2676 rows x 4 columns]

In [18]:
df_text1.value_counts('judgment')

judgment
Y    2614
N      62
dtype: int64

In [19]:
df_text2 = trainintuitive_data[(trainintuitive_data['judgment'] == 'Y') | (trainintuitive_data['judgment']=='N')]

In [20]:
df_text2

source                  name    id judgment
0     intuitive                Asthma     1        N
1     intuitive                Asthma     2        Y
2     intuitive                Asthma     4        N
3     intuitive                Asthma     6        N
4     intuitive                Asthma    15        N
...         ...                   ...   ...      ...
8616  intuitive  Venous Insufficiency  1240        N
8617  intuitive  Venous Insufficiency  1244        Y
8618  intuitive  Venous Insufficiency  1245        N
8619  intuitive  Venous Insufficiency  1248        N
8620  intuitive  Venous Insufficiency  1249        Y

[8598 rows x 4 columns]

In [21]:
df_text2.value_counts('judgment')

judgment
N    5887
Y    2711
dtype: int64

<bold><italic> we have extracted all the clinical records having labels ‘‘Y’’ or ‘‘N’’ from the textual andintuitive judgment documents into df_text1 (train textual) & df_text2(intuitive) </italic></bold>

df_text1['name'].str.lower()

df_text2['name'].str.lower()

df_text1

dftt = traintext_data.copy(deep=True)

dfit = trainintuitive_data.copy(deep=True)

dftt['name'] = dftt['name'].str.lower()

dfit['name'] = dfit['name'].str.lower()

In [22]:
train_records = pd.read_csv('trainrecords.csv')

In [23]:
train_records

id                                               text
0       1  490646815 | WMC | 31530471 | | 9629480 | 11/23...
1       2  159644670 | VH | 60656526 | | 6334749 | 11/29/...
2       4  368346277 | EMH | 64927307 | | 815098 | 3/29/1...
3       6  018858680 | AOH | 80239131 | | 9725704 | 11/4/...
4      13  908761918 | MMC | 45427009 | | 0927689 | 5/26/...
..    ...                                                ...
606  1244  878570799 | VC | 87322051 | | 3512149 | 10/9/2...
607  1245  389697639 | CHH | 44087718 | | 300648 | 6/9/19...
608  1246  559327548 | BCVHC | 64379841 | | 265288 | 5/5/...
609  1248  997921897 | MMH | 62172046 | | 8671687 | 5/21/...
610  1249  868276463 | CH | 49907736 | | 0475555 | 10/4/2...

[611 rows x 2 columns]

In [24]:
train_records['text'].iloc[1]

"159644670 | VH | 60656526 | | 6334749 | 11/29/2005 12:00:00 AM | Discharge Summary | Signed | DIS | Admission Date: 9/10/2005 Report Status: Signed\n\nDischarge Date:\nATTENDING: DROEGMILLER , STANTON MD\nDATE OF DISCHARGE: Not known at the time of this interim\ndictation.\nSERVICE: List Medical Center .\nPRIMARY DIAGNOSIS: Congestive heart failure.\nSECONDARY DIAGNOSES: Postpartum cardiomyopathy status post\nmitral valve repair , status post tricuspid valve repair , status\npost ICD/pacer placement , asthma , and anemia.\nHISTORY OF PRESENT ILLNESS: This is a 33-year-old female with a\nhistory of postpartum cardiomyopathy ( EF 15% to 20% ) , status post\npacer/ICD placement , status post mitral valve\nreplacement/tricuspid valve replacement , who presents with\nabdominal pain , decreased appetite , nausea , vomiting , and\noccasional chest pain. The patient has had multiple recent\nadmissions to H/seases Medical Center .  In October 2005 , the\npatient presented with chest pain and w

In [25]:
train_records.shape

(611, 2)

In [26]:
type(train_records)

pandas.core.frame.DataFrame

In [27]:
len(train_records)

611

In [28]:
train_records = train_records.apply(lambda x: x.astype(str).str.lower())

In [29]:
type(train_records)

pandas.core.frame.DataFrame

In [30]:
train_records = pd.DataFrame({'sentences': train_records['text']})
train_records['tokenized_sents'] = train_records.apply(lambda row: nltk.word_tokenize(row['sentences']), axis=1)

In [31]:
train_records

sentences  \
0    490646815 | wmc | 31530471 | | 9629480 | 11/23...   
1    159644670 | vh | 60656526 | | 6334749 | 11/29/...   
2    368346277 | emh | 64927307 | | 815098 | 3/29/1...   
3    018858680 | aoh | 80239131 | | 9725704 | 11/4/...   
4    908761918 | mmc | 45427009 | | 0927689 | 5/26/...   
..                                                 ...   
606  878570799 | vc | 87322051 | | 3512149 | 10/9/2...   
607  389697639 | chh | 44087718 | | 300648 | 6/9/19...   
608  559327548 | bcvhc | 64379841 | | 265288 | 5/5/...   
609  997921897 | mmh | 62172046 | | 8671687 | 5/21/...   
610  868276463 | ch | 49907736 | | 0475555 | 10/4/2...   

                                       tokenized_sents  
0    [490646815, |, wmc, |, 31530471, |, |, 9629480...  
1    [159644670, |, vh, |, 60656526, |, |, 6334749,...  
2    [368346277, |, emh, |, 64927307, |, |, 815098,...  
3    [018858680, |, aoh, |, 80239131, |, |, 9725704...  
4    [908761918, |, mmc, |, 45427009, |, |, 0927689...  
..                                                 ...  
606  [878570799, |, vc, |, 87322051, |, |, 3512149,...  
607  [389697639, |, chh, |, 44087718, |, |, 300648,...  
608  [559327548, |, bcvhc, |, 64379841, |, |, 26528...  
609  [997921897, |, mmh, |, 62172046, |, |, 8671687...  
610  [868276463, |, ch, |, 49907736, |, |, 0475555,...  

[611 rows x 2 columns]

In [32]:
train_records['tokenized_sents'].iloc[1]

['159644670',
 '|',
 'vh',
 '|',
 '60656526',
 '|',
 '|',
 '6334749',
 '|',
 '11/29/2005',
 '12:00:00',
 'am',
 '|',
 'discharge',
 'summary',
 '|',
 'signed',
 '|',
 'dis',
 '|',
 'admission',
 'date',
 ':',
 '9/10/2005',
 'report',
 'status',
 ':',
 'signed',
 'discharge',
 'date',
 ':',
 'attending',
 ':',
 'droegmiller',
 ',',
 'stanton',
 'md',
 'date',
 'of',
 'discharge',
 ':',
 'not',
 'known',
 'at',
 'the',
 'time',
 'of',
 'this',
 'interim',
 'dictation',
 '.',
 'service',
 ':',
 'list',
 'medical',
 'center',
 '.',
 'primary',
 'diagnosis',
 ':',
 'congestive',
 'heart',
 'failure',
 '.',
 'secondary',
 'diagnoses',
 ':',
 'postpartum',
 'cardiomyopathy',
 'status',
 'post',
 'mitral',
 'valve',
 'repair',
 ',',
 'status',
 'post',
 'tricuspid',
 'valve',
 'repair',
 ',',
 'status',
 'post',
 'icd/pacer',
 'placement',
 ',',
 'asthma',
 ',',
 'and',
 'anemia',
 '.',
 'history',
 'of',
 'present',
 'illness',
 ':',
 'this',
 'is',
 'a',
 '33-year-old',
 'female',
 'with',
 

In [33]:
tr = train_records.copy(deep=True)

In [34]:
tr

sentences  \
0    490646815 | wmc | 31530471 | | 9629480 | 11/23...   
1    159644670 | vh | 60656526 | | 6334749 | 11/29/...   
2    368346277 | emh | 64927307 | | 815098 | 3/29/1...   
3    018858680 | aoh | 80239131 | | 9725704 | 11/4/...   
4    908761918 | mmc | 45427009 | | 0927689 | 5/26/...   
..                                                 ...   
606  878570799 | vc | 87322051 | | 3512149 | 10/9/2...   
607  389697639 | chh | 44087718 | | 300648 | 6/9/19...   
608  559327548 | bcvhc | 64379841 | | 265288 | 5/5/...   
609  997921897 | mmh | 62172046 | | 8671687 | 5/21/...   
610  868276463 | ch | 49907736 | | 0475555 | 10/4/2...   

                                       tokenized_sents  
0    [490646815, |, wmc, |, 31530471, |, |, 9629480...  
1    [159644670, |, vh, |, 60656526, |, |, 6334749,...  
2    [368346277, |, emh, |, 64927307, |, |, 815098,...  
3    [018858680, |, aoh, |, 80239131, |, |, 9725704...  
4    [908761918, |, mmc, |, 45427009, |, |, 0927689...  
..                                                 ...  
606  [878570799, |, vc, |, 87322051, |, |, 3512149,...  
607  [389697639, |, chh, |, 44087718, |, |, 300648,...  
608  [559327548, |, bcvhc, |, 64379841, |, |, 26528...  
609  [997921897, |, mmh, |, 62172046, |, |, 8671687...  
610  [868276463, |, ch, |, 49907736, |, |, 0475555,...  

[611 rows x 2 columns]

In [35]:
l=[]
for i in tr['tokenized_sents'] :
    for char in i:
        if not str(char).isalnum():
            l.append(char)

In [36]:
l

['|',
 '|',
 '|',
 '|',
 '|',
 '11/23/2006',
 '12:00:00',
 '|',
 '|',
 '|',
 '|',
 ':',
 '11/23/2006',
 ':',
 ':',
 '6/20/2006',
 ':',
 ',',
 'm.d',
 '.',
 ':',
 '.',
 ':',
 '.',
 ':',
 ',',
 ',',
 ',',
 '.',
 ':',
 '86-year-old',
 ',',
 ',',
 '%',
 '%',
 '.',
 ',',
 ',',
 '.',
 'dr.',
 '.',
 'band-like',
 '.',
 '.',
 'pre-admission',
 ':',
 'p.o',
 '.',
 'b.i.d',
 '.',
 ',',
 'q.p.m',
 '.',
 ',',
 'units/4',
 'units/5',
 't.i.d',
 '.',
 ',',
 'b.i.d',
 '.',
 ',',
 'b.i.d',
 '.',
 ',',
 ',',
 ',',
 ',',
 'p.o',
 '.',
 ',',
 '.',
 ':',
 ',',
 'failure/diuresis/renal',
 'disease/early',
 ';',
 ';',
 ';',
 '%',
 '%',
 ';',
 ';',
 ';',
 ';',
 '.',
 ':',
 '.',
 ':',
 ',',
 ',',
 '.',
 ':',
 '.',
 ':',
 '96.7',
 ',',
 ',',
 '153/74',
 ',',
 ',',
 '%',
 '.',
 '.',
 '.',
 '.',
 '.',
 ',',
 '3/6',
 '.',
 '.',
 '.',
 '.',
 ',',
 '.',
 ':',
 ',',
 ',',
 ',',
 '.',
 'x-ray',
 ',',
 '.',
 'x-ray',
 ',',
 '1.',
 '.',
 '2.',
 '.',
 '3.',
 '.',
 '10/24/06',
 ',',
 ';',
 'h.',
 ';',
 '.',
 '7/5/06',
 '

In [37]:
len(l)

191760

In [38]:
l=set(l)
len(l)

9511

In [39]:
l=sorted(l)

In [40]:
alpha = 'abcdefghijklmnopqrstuvwxyz'

In [41]:
num = range(1,11,1)

In [42]:
num

range(1, 11)

In [43]:
c=False
for i in l:
    for j in i:
        if (j in alpha) or (j in num):
            c=True
    if(c==True):
        l.remove(i)

In [44]:
len(l)

4765

In [45]:
l

['!',
 '#',
 '%',
 '&',
 "'",
 "''",
 "'00",
 "'02",
 "'03",
 "'04",
 "'70",
 "'73",
 "'74",
 "'76",
 "'83",
 "'88",
 "'89",
 "'91",
 "'92",
 "'93",
 "'95",
 "'97",
 "'99",
 "'non-diagnostic",
 "'pounding",
 "'s",
 "'ve",
 ')',
 '+',
 '++chills',
 '++rbcs',
 '+/-',
 '+1',
 '+2',
 '+2/4',
 '+60',
 '+8cm',
 '+bnp',
 '+brbpr',
 '+cardiomegaly',
 '+cmv',
 '+cv',
 '+diaphoresis',
 '+dipyrimadole',
 '+dry',
 '+edema',
 '+ekg',
 '+epigastric',
 '+ext',
 '+family',
 '+fhx',
 '+h',
 '+head',
 '+hypoflexia',
 '+increased',
 '+l',
 '+lh',
 '+lupus',
 '+lvh',
 '+minimal',
 '+nitrites',
 '+orthopnea',
 '+palps',
 '+pnd',
 '+psych',
 '+repetition',
 '+s4',
 '+smoking',
 '+supraspinatus',
 '+thrill',
 '+tobacco',
 '+trop',
 '+uri',
 '+ventral',
 '-',
 '--',
 '-04',
 '-1',
 '-1500.',
 '-1l',
 '-25',
 '-2l',
 '-4',
 '-5.4',
 '-500',
 '-60',
 '-7',
 '-a',
 '-advair',
 '-allopurinol',
 '-anticoagulation',
 '-bph',
 '-calcium/vit',
 '-can',
 '-cbc',
 '-check',
 '-chf',
 '-combivent',
 '-consider',
 '-cont

In [46]:
import re
for i in l:
    x = re.findall("[a-zA-Z]", i)
    if(bool(x)):
        l.remove(i)

In [47]:
len(l)

3783

In [48]:
l

['!',
 '#',
 '%',
 '&',
 "'",
 "''",
 "'00",
 "'02",
 "'03",
 "'04",
 "'70",
 "'73",
 "'74",
 "'76",
 "'83",
 "'88",
 "'89",
 "'91",
 "'92",
 "'93",
 "'95",
 "'97",
 "'99",
 "'pounding",
 "'ve",
 ')',
 '+',
 '++rbcs',
 '+/-',
 '+1',
 '+2',
 '+2/4',
 '+60',
 '+bnp',
 '+cardiomegaly',
 '+cv',
 '+dipyrimadole',
 '+edema',
 '+epigastric',
 '+family',
 '+h',
 '+hypoflexia',
 '+l',
 '+lupus',
 '+minimal',
 '+orthopnea',
 '+pnd',
 '+repetition',
 '+smoking',
 '+thrill',
 '+trop',
 '+ventral',
 '-',
 '--',
 '-04',
 '-1',
 '-1500.',
 '-25',
 '-4',
 '-5.4',
 '-500',
 '-60',
 '-7',
 '-advair',
 '-anticoagulation',
 '-calcium/vit',
 '-cbc',
 '-chf',
 '-consider',
 '-ct/a',
 '-done',
 '-drink',
 '-efficacy',
 '-etoh',
 '-fluid',
 '-glucotrol',
 '-hemodialysis',
 '-hospital',
 '-hyperlipidemia',
 '-imdur',
 '-ischemia',
 '-l',
 '-lovenox',
 '-mibi',
 '-nephrocaps',
 '-p',
 '-pmhx',
 '-pump',
 '-restarted',
 '-rheum',
 '-s',
 '-sh',
 '-stop',
 '-topical',
 '-unclear',
 '-you',
 '.',
 '...',
 '.03',
 

In [49]:
r = "^[+][a-z]"

In [50]:
import re
li=[]
for i in l:
    li = re.findall(r,i)
    if(bool(li)):
        l.remove(i)

In [51]:
len(l)

3773

In [52]:
l

['!',
 '#',
 '%',
 '&',
 "'",
 "''",
 "'00",
 "'02",
 "'03",
 "'04",
 "'70",
 "'73",
 "'74",
 "'76",
 "'83",
 "'88",
 "'89",
 "'91",
 "'92",
 "'93",
 "'95",
 "'97",
 "'99",
 "'pounding",
 "'ve",
 ')',
 '+',
 '++rbcs',
 '+/-',
 '+1',
 '+2',
 '+2/4',
 '+60',
 '+cardiomegaly',
 '+dipyrimadole',
 '+epigastric',
 '+h',
 '+l',
 '+minimal',
 '+pnd',
 '+smoking',
 '+trop',
 '-',
 '--',
 '-04',
 '-1',
 '-1500.',
 '-25',
 '-4',
 '-5.4',
 '-500',
 '-60',
 '-7',
 '-advair',
 '-anticoagulation',
 '-calcium/vit',
 '-cbc',
 '-chf',
 '-consider',
 '-ct/a',
 '-done',
 '-drink',
 '-efficacy',
 '-etoh',
 '-fluid',
 '-glucotrol',
 '-hemodialysis',
 '-hospital',
 '-hyperlipidemia',
 '-imdur',
 '-ischemia',
 '-l',
 '-lovenox',
 '-mibi',
 '-nephrocaps',
 '-p',
 '-pmhx',
 '-pump',
 '-restarted',
 '-rheum',
 '-s',
 '-sh',
 '-stop',
 '-topical',
 '-unclear',
 '-you',
 '.',
 '...',
 '.03',
 '.076',
 '.10-0.8',
 '.125',
 '.14',
 '.164',
 '.18/.08/.42',
 '.2.',
 '.25',
 '.3',
 '.44',
 '.467',
 '.5',
 '.6',
 '.66',

In [53]:
import re

p = "[+|-][a-z][0-9][A-z][0-9]+"
for i in l: 
    if not bool(re.findall(p,i)):
        l.remove(i)
        

In [54]:
len(l)

1886

In [55]:
l

['#',
 '&',
 "''",
 "'02",
 "'04",
 "'73",
 "'76",
 "'88",
 "'91",
 "'93",
 "'97",
 "'pounding",
 ')',
 '++rbcs',
 '+1',
 '+2/4',
 '+cardiomegaly',
 '+epigastric',
 '+l',
 '+pnd',
 '+trop',
 '--',
 '-1',
 '-25',
 '-5.4',
 '-60',
 '-advair',
 '-calcium/vit',
 '-chf',
 '-ct/a',
 '-drink',
 '-etoh',
 '-glucotrol',
 '-hospital',
 '-imdur',
 '-l',
 '-mibi',
 '-p',
 '-pump',
 '-rheum',
 '-sh',
 '-topical',
 '-you',
 '...',
 '.076',
 '.125',
 '.164',
 '.2.',
 '.3',
 '.467',
 '.6',
 '.75',
 '.9',
 '/18',
 '/hr',
 '0-0487435',
 '0-1.5',
 '0-5',
 '0-7283079',
 '0-9620528',
 '0.0.',
 '0.00.',
 '0.02.',
 '0.04',
 '0.06',
 '0.08',
 '0.11',
 '0.125',
 '0.133',
 '0.17',
 '0.187',
 '0.2',
 '0.219',
 '0.24',
 '0.250',
 '0.29',
 '0.3/0.2',
 '0.333',
 '0.375',
 '0.4-0.8',
 '0.401.',
 '0.44',
 '0.46',
 '0.5.',
 '0.59',
 '0.60',
 '0.67',
 '0.71.',
 '0.8',
 '0.85.',
 '0.89.',
 '0.91',
 '0.98',
 '0/5',
 '000.',
 '011-31-75-9',
 '01:05',
 '01:22',
 '01:29',
 '01:44',
 '02.',
 '026-24-73-5',
 '02:00',
 '02:12'

In [56]:
p = "[+|-][+]*[a-z][0-9][A-z][0-9]+"
for i in l: 
    if not bool(re.findall(p,i)):
        l.remove(i)

In [57]:
len(l)

943

In [58]:
l

['&',
 "'02",
 "'73",
 "'88",
 "'93",
 "'pounding",
 '++rbcs',
 '+2/4',
 '+epigastric',
 '+pnd',
 '--',
 '-25',
 '-60',
 '-calcium/vit',
 '-ct/a',
 '-etoh',
 '-hospital',
 '-l',
 '-p',
 '-rheum',
 '-topical',
 '...',
 '.125',
 '.2.',
 '.467',
 '.75',
 '/18',
 '0-0487435',
 '0-5',
 '0-9620528',
 '0.00.',
 '0.04',
 '0.08',
 '0.125',
 '0.17',
 '0.2',
 '0.24',
 '0.29',
 '0.333',
 '0.4-0.8',
 '0.44',
 '0.5.',
 '0.60',
 '0.71.',
 '0.85.',
 '0.91',
 '0/5',
 '011-31-75-9',
 '01:22',
 '01:44',
 '026-24-73-5',
 '02:12',
 '02:38',
 '031-08-99-2',
 '03:12',
 '03:42',
 '04:00',
 '04:55',
 '05:17',
 '066-44-02-7',
 '06:26',
 '07:11',
 '089-62-23-0',
 '094-07-84-0',
 '09:46',
 '1-1-1/2',
 '1-2/10',
 '1-3wks',
 '1-7858650',
 '1.',
 '1.025',
 '1.1.',
 '1.2-2.3',
 '1.29.',
 '1.38.',
 '1.49',
 '1.5l',
 '1.61.',
 '1.7-3.3',
 '1.8-2.5',
 '1.9-2.0',
 '1.only',
 '1/1/2004',
 '1/10/1993',
 '1/10/97',
 '1/12',
 '1/12/2006',
 '1/13/2004',
 '1/15/03',
 '1/15/92',
 '1/17/03',
 '1/17/95',
 '1/18/1998',
 '1/19/03',